### PR TITLE
Evolution of YOLO

### PR DESCRIPTION
**Pull request by AntonisCSt**
This PR addresses  #23 Issue of Unit's 2 Yolo chapter #20

### CODE REVIEW

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+### **YOLOv2 (2016)** 
+
+After a year of releasing the first version, YOLOv2[1] came out. The improvements focused on both accuracy and speed but also dealt with the localization problem. Many improvements were made (image1). First, YOLOv2 replaced YOLOv1’s backbone architecture with Darknet-19, which is a variant of the Darknet architecture. Darknet-19 is lighter than the previous version’s backbone and it consists of 19 convolutional layers followed by max-pooling layers. This led to YOLOv2 having the ability to capture more information. Also, it included batch normalization to all convolutional layers removed the dropout layers dealing with the overfitting problem, and increased the mAP. It also introduced anchor boxes, which added prior knowledge to the detected width and height of the detected boxes (specifically they used). Additionally, to deal with the poor localization YOLOv2 predicted the class and objects for every anchor box and as previously grid (now 13x13). So we have a maximum (for 5 anchor boxes) of 13x13x5 = 845 boxes.
```
Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]: what will be the `image1` ?

Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]:
```suggestion
After a year of releasing the first version, YOLOv2[1] came out. The improvements focused on both accuracy and speed but also dealt with the localization problem. Many improvements were made (image1). First, YOLOv2 replaced YOLOv1’s backbone architecture with Darknet-19, which is a variant of the Darknet architecture. Darknet-19 is lighter than the previous version’s backbone and it consists of 19 convolutional layers followed by max-pooling layers. This led to YOLOv2 having the ability to capture more information. Also, it included batch normalization to all convolutional layers and hence, removed the dropout layers dealing with the overfitting problem increasing the mAP. It also introduced the concept of anchor boxes, which added prior knowledge to the detected width and height of the detected boxes (specifically they used). Additionally, to deal with the poor localization YOLOv2 predicted the class and objects for every anchor box and as previously grid (now 13x13). So we have a maximum (for 5 anchor boxes) of 13x13x5 = 845 boxes.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+### **YOLOv3 (2018)**
+
+YOLOv3[2] again improved significantly the detection speed and accuracy by replacing the Darknet-19 architecture with the more complex but efficient Darknet-53. Also, it dealt more with the localization problem by using three different scales for object detection (13x13, 26x26, and 52x52 grids). This helped find objects with different sizes in the same area. This increased the bounding boxes to: 13 x 13 x 3 + 26 x 26 x 3 + 52 x 52 x 3 = 10,647. The Non-Maximum Suppression (NMS) was still used to filter out redundant overlapping boxes.
```
Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]:
```suggestion
YOLOv3[2] again improved significantly the detection speed and accuracy by replacing the Darknet-19 architecture with the more complex but efficient Darknet-53. Also, it dealt with the localization problem better by using three different scales for object detection (13x13, 26x26, and 52x52 grids). This helped finding the objects with different sizes in the same area. It increased the bounding boxes to: 13 x 13 x 3 + 26 x 26 x 3 + 52 x 52 x 3 = 10,647. The Non-Maximum Suppression (NMS) was still used to filter out redundant overlapping boxes.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+### **YOLOv4 (2020)**
+
+Back in 2020, YOLOv4[3] became one of the best detection models in terms of speed and accuracy. The authors changed again the backbone architecture to the faster and better accuracy CSPDarknet53[4]. An important improvement of this version was the optimization for efficient resource utilization, making it suitable for deployment on various hardware platforms, including edge devices. Also, It included a number of augmentations before training that further improved the model prediction accuracy on custom data.
```
Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]: what exact accuracy? by custom data you mean new dataset?

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+### **YOLOv6 (2022)**
+
+The next version, YOLOv6[6][7], was released by the Meituan Vision AI Department under the article title: "YOLOv6: A Single-Stage Object Detection Framework for Industry." This team made further improvements in terms of speed and accuracy by focusing on five aspects:
+1) Reparameterization using the RepVGG technique. Which is a modified version of VGG with skip connections. During inference, these connections are fused to improve the speed.
+2) Quantization of reparameterization-based detectors. Which added blocks that are called Rep-PANs.
+3) Recognition of the importance of considering different hardware costs and capabilities for model deployment.
```
Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]:
```suggestion
3) Recognition of the importance of considering different hardware costs and capabilities for model deployment.
```
I do not understand what you meant

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+4) Further verification of label assignment and loss function design after architectural improvements. YOLOv6 uses Varifocal Loss for Classification. For Box Regression, it uses IoU Series Loss and Distribution Focal Loss.
```
Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]:
```suggestion
4) Introducttion to new types of loss functions such as, Varifocal Loss for Classification. IoU Series Loss  for Boundig Box Regression, and Distribution Focal Loss.
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+### **YOLOv8 (2023)**
+
+YOLOv8[11], developed by Ultralytics in 2023 became again the new SOTA. It introduced improvements on the backbone and neck, alongside an anchor-free approach which eliminates the need for predefined anchor boxes. Instead, predictions are made directly. This version supports a wide range of vision tasks including classification, segmentation, and pose estimation. Additionally, YOLOv8 has scaling capabilities with pre-trained models available in various sizes: nano, small, medium, large, and extra-large, and can easily fine-tuned on custom datasets. 
+
+### **YOLOv9 (2024)**
+
+YOLOv9 was released with the paper name: “YOLOv9: Learning What You Want to Learn Using Programmable Gradient Information”[12][13] by a similar team of YOLOv7 and YOLOv4 authors. This paper highlights the issue of information loss that existing methods and architectures have during layer-by-layer feature extraction and spatial transformation. To address this issue, the authors proposed:
```
Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]:
```suggestion
YOLOv9 was released with the paper name: “YOLOv9: Learning What You Want to Learn Using Programmable Gradient Information”[12][13] by the same authors of YOLOv7 and YOLOv4 . This paper highlights the issue of information loss that existing methods and architectures have during layer-by-layer feature extraction and spatial transformation. To address this issue, the authors proposed:
```

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+* Concept of programmable gradient information (PGI) to cope with the various changes
+required by deep networks to achieve multiple objectives.
+* Generalized Efficient Layer Aggregation Network (GELAN), a new lightweight network architecture that achieves better parameter utilization than the current methods without sacrificing computational efficiency.
+
+With these changes, YOLOv9 set new benchmarks on the MS COCO.
```
Comment (sezan92) [inline chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]:
```suggestion
With these changes, YOLOv9 set new benchmarks on the MS COCO challenge
```

**Review by sezan92 — COMMENTED** [review]
commented

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx (hunk 1 above)
Comment (AntonisCSt) [reply chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]: It was in the first draft that I was making. I did not use it eventually, so it is a mistake. Thanks

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx (hunk 3 above)
Comment (AntonisCSt) [reply chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]: _**what exact accuracy?**_ In object detection benchmarks like COCO dataset

 **_by custom data you mean new dataset?_** Yes, but I will change that and mention that it helps in the generalization of the model.

File: chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx (hunk 4 above)
Comment (AntonisCSt) [reply chapters/en/Unit 2 - Convolutional Neural Networks/yolo.mdx]: As the [paper ](https://arxiv.org/pdf/2209.02976) mentions: 

_Previous works tend to pay less attention
to deployment, whose latencies are commonly compared
on high-cost machines like V100. There is a hardware gap
when it comes to real serving environment. Typically, lowpower GPUs like Tesla T4 are less costly and provide rather good inference performance._

So in their work they made tests with such hardware which did not happen in the previous works. I will add a similar explanation. Thanks

**Review by JvThunder — APPROVED** [review]
Very well written! However, I have suggestions that you can add. 
YOLOv1: It’s worth mentioning that YOLOv1 used a single neural network to predict bounding boxes and class probabilities directly from full images in one evaluation.
Maybe can also explain briefly what YOLO actually does.
The original YOLO model introduced the concept of using a single convolutional neural network (CNN) that processes the entire image at once, dividing it into an SxS grid. Each grid cell predicts bounding boxes and class probabilities directly. 

Other than that, it looks great!

**Comment by AntonisCSt** [comment]
> Very well written! However, I have suggestions that you can add. YOLOv1: It’s worth mentioning that YOLOv1 used a single neural network to predict bounding boxes and class probabilities directly from full images in one evaluation. Maybe can also explain briefly what YOLO actually does. The original YOLO model introduced the concept of using a single convolutional neural network (CNN) that processes the entire image at once, dividing it into an SxS grid. Each grid cell predicts bounding boxes and class probabilities directly.
> 
> Other than that, it looks great!

Thanks added some changes in the introduction!

**Review by sezan92 — APPROVED** [review]

**Pull request merged**